src/utils/hdfs_mods.py

```diff
@@ -165,3 +166,95 @@ def hdfs_write_feather(filepath, df):
     hdfs_logger.info(f"Dataframe written to {filepath} as feather file")
 
     return True
+
+def _perform(command, shell: bool = False, str_output: bool = False, ignore_error: bool = False, full_out=False):
+    """
+    Run shell command in subprocess returning exit code or full string output.
+    _perform() will build the command that will be put into HDFS.
+    This will also be used for the functions below.
+    Parameters
+    ----------
+    shell
+        If true, the command will be executed through the shell.
+        See subprocess.Popen() reference.
+    str_output
+        output exception as string
+    ignore_error
+    """
+    process = subprocess.Popen(command, shell=shell, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+
+    if str_output:
+        if stderr and not ignore_error:
+            raise Exception(stderr.decode("UTF-8").strip("\n"))
+        if full_out:
+            return stdout
+        return stdout.decode("UTF-8").strip("\n")
+
+    return process.returncode == 0
+
+def hdfs_delete_file(path: str):
+    """
+    Delete a file. Uses 'hadoop fs -rm'.
+
+    Returns
+    -------
+    True for successfully completed operation. Else False.
+    """
+    command = ["hadoop", "fs", "-rm", path]
+    return _perform(command)
+
+def hdfs_md5sum(path: str):
+    """
+    Get md5sum of a specific file on HDFS.
+    """
+    return _perform(f"hadoop fs -cat {path} | md5sum", shell=True, str_output=True, ignore_error=True).split(" ")[0]
```

Comment:
Could ignore_error be changed to False and errors handled, maybe with try except
```suggestion
try:
    return _perform(f"hadoop fs -cat {path} | md5sum", shell=True, str_output=True, ignore_error=True).split(" ")[0]
except Exception as e:
    logging.error(f"Error reading header for path {path}: {e}")
    raise
```